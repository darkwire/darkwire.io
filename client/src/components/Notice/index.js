import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

const Notice = props => (
  <div>
    <div className={classNames({
      info: props.level === 'info',
      warning: props.level === 'warning',
      danger: props.level === 'danger',
    })}>
      {props.children}
    </div>
  </div>
)

Notice.defaultProps = {
  level: 'info',
}

Notice.propTypes = {
  children: PropTypes.node.isRequired,
  level: PropTypes.string,
}

export default Notice
