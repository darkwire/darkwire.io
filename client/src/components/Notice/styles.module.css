.info { 
  color: white
}
.warning {
  color: yellow
}

.danger {
  color: gold
}
