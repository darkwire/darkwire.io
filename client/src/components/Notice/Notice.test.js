import React from 'react';
import { render } from '@testing-library/react';
import Notice from '.';

test('Notice component is displaying', async () => {
  const { asFragment } = render(
    <Notice level={'warning'}>
      <div>Hello world</div>
    </Notice>,
  );

  expect(asFragment()).toMatchSnapshot();
});
