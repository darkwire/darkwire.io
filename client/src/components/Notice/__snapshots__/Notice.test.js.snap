// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notice component is displaying 1`] = `
<DocumentFragment>
  <div>
    <div
      class="warning"
    >
      <div>
        Hello world
      </div>
    </div>
  </div>
</DocumentFragment>
`;
